### PR TITLE
Added infection and improved some tests

### PR DESCRIPTION
**Pull request by alfredbez**

### CODE REVIEW

File: tests/Unit/DataType/BoolFilterTest.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        $this->assertTrue((new BoolFilter())->equals());
+    }
+
+    /**
+     * @dataProvider boolDataProvider
+     *
+     * @param mixed $userInput
+     * @param mixed $returnValue
+     */
+    public function testReturnsGivenEqualValue($userInput, $returnValue): void
```
Comment (flow-control) [inline tests/Unit/DataType/BoolFilterTest.php]:
```suggestion
     */
    public function testReturnsGivenEqualValue(bool $userInput, bool $returnValue): void
```

File: tests/Unit/DataType/FloatFilterTest.php [hunk 2]
```diff
@@ -21,6 +21,12 @@ public function testThrowsExceptionOnNoInput(): void
         FloatFilter::fromUserInput();
     }
 
+    public function testGivesEqualParameterIfSet(): void
+    {
+        $filter = new FloatFilter(1.0);
+        $this->assertEquals(1.0, $filter->equals());
```
Comment (flow-control) [inline tests/Unit/DataType/FloatFilterTest.php]:
```suggestion
        $this->assertSame(1.0, $filter->equals());
```

File: tests/Unit/Framework/RequestReaderTest.php [hunk 3]
```diff
@@ -118,4 +120,22 @@ public function testGetGraphQLRequestDataWithInputRequest(): void
...
     }
+
+    public function testGetGraphQLRequestDataWithInputRequestWithoutJson(): void
+    {
+        $requestReader             = new RequestReader();
+        $_SERVER['CONTENT_TYPE']   = 'text/plain';
+        $_REQUEST['query']         = 'query {token_}';
+        $_REQUEST['variables']     = '{"foo":"bar"}';
+        $_REQUEST['operationName'] = 'operation_name';
+        $this->assertEquals(
```
Comment (flow-control) [inline tests/Unit/Framework/RequestReaderTest.php]:
```suggestion
        $this->assertSame(
```

File: tests/Unit/Framework/RequestReaderTest.php [hunk 4]
```diff
@@ -118,4 +120,22 @@ public function testGetGraphQLRequestDataWithInputRequest(): void
...
+            [
+                'query'         => 'query {token_}',
+                'variables'     => ['foo' => 'bar'],
+                'operationName' => 'operation_name',
+            ],
+            $requestReader->getGraphQLRequestData()
+        );
+        unset($_SERVER['CONTENT_TYPE']);
```
Comment (flow-control) [inline tests/Unit/Framework/RequestReaderTest.php]:
```suggestion
        unset($_SERVER['CONTENT_TYPE']);
        unset($_REQUEST['query']);
        unset($_REQUEST['variables']);
        unset($_REQUEST['operationName']);
```

File: .travis.yml [hunk 5]
```diff
@@ -30,3 +30,4 @@ script:
     - composer test:lint-fixer
     - composer test:static
     - composer test:unit
+    - composer test:infection
```
Comment (flow-control) [inline .travis.yml]: Could you somehow update this to only run on PHP >= 7.2? I think it is okay to ignore PHP 7.1 in this case

Comment (alfredbez) [reply .travis.yml]: I downgraded infection to an older version which is compatible with PHP 7.1

**Comment by alfredbez** [comment]
I've fixed the open things and rebased it, build is green now for all PHP versions.

> One thing I will have a deeper look into is `tests/Unit/Controller/LoginTest.php`. It somehow bothers me ;-) Mainly I will investigate if there is a better option then to have these mocks.

I basically copied it from [here](https://github.com/OXID-eSales/graphql-base-module/blob/795b056ce5cf52221d4580875e66ccca28476c6d/tests/Unit/Service/AuthenticationServiceTest.php#L39-L46), maybe we can find a cleaner solution for both cases.

**Comment by flow-control** [comment]
> > One thing I will have a deeper look into is `tests/Unit/Controller/LoginTest.php`. It somehow bothers me ;-) Mainly I will investigate if there is a better option then to have these mocks.
> 
> I basically copied it from [here](https://github.com/OXID-eSales/graphql-base-module/blob/795b056ce5cf52221d4580875e66ccca28476c6d/tests/Unit/Service/AuthenticationServiceTest.php#L39-L46), maybe we can find a cleaner solution for both cases.

Yeah I did this, and I dont like it ;-)

**Comment by flow-control** [comment]
I don't like the alternative either :wink: 
LGTM and thanks again for this contribution!

**Pull request merged**